drivers/bmx280/bmx280.c

```diff
@@ -225,13 +219,15 @@ static int read_calibration_data(bmx280_t* dev)
     uint8_t offset = 0x88;
 
     memset(buffer, 0, sizeof(buffer));
-    nr_bytes = i2c_read_regs(dev->params.i2c_dev, dev->params.i2c_addr, offset,
-                             buffer, nr_bytes_to_read);
-    if (nr_bytes != nr_bytes_to_read) {
+    i2c_acquire(dev->params.i2c_dev);
+    result = i2c_read_regs(dev->params.i2c_dev, dev->params.i2c_addr, offset,
+                           buffer, nr_bytes_to_read, 0);
+    i2c_release(dev->params.i2c_dev);
+    if (result != I2C_ACK) {
```

Comment:
`I2C_ACK` 'return code was removed from the API, prefer a check like `!=` or `<0`.